### ISSUE TITLE
Using HackRF amplifier

### ISSUE THREAD

**Issue by Matioupi** [martinmarinov/rtl_tcp_andro-#26]
Hello,
i'm trying to use at same spot, dump1090 works fine on my computer).

My guess is that the internal hackRF amplifier should be activated, but I can't figure out how to do it from android, would it be possible to have a "config file mode" to force some hackrf settings not handled by sdrTcpArguments ?

regards

**Comment by Matioupi** [comment]
Justfor the follow-up, I've been recompiling your app by adding
`hackrf.setAmp(true);`
in HackRfTcp.java executCommand method

With this mod, ADSB Flight Tracker app works much better, which seems to confirm my previous thought (but the hack is quite dirty...) Having logics to control the various gain stages of the hackRF from the GUI would be better.

**Comment by Enthusiastic-Coder** [comment]
Hi Matioupi,

Not sure - is this a general comment or query to me?

Thanks.

**Comment by Enthusiastic-Coder** [comment]
Hi Matioupi,

i think i pass 3 parameters ; port no, auto gain and frequency via Android intent.
Are you asking for a 4th?
"hackrf.setAmp(true);"

**Comment by Matioupi** [comment]
Right, the hackrf has an internal amplifier that can be enabled or not, and my guess is that supporting hackrf would also need supporting the various hackRF gain stages through the passed arguments.
If you could add this, that would be really great ! (and also make persistent the default settings by the user). By the way, this is a very helpfull piece of software !

**Comment by Enthusiastic-Coder** [comment]
Hi Matioupi

Appreciate your feedback!
The arguments I currently pass are as follows:

public void startIntent(int port, int freq, int samplerate) {

        Intent intent = new Intent(Intent.ACTION_VIEW).setData(
            Uri.parse("iqsrc://-a 0.0.0.0 -p "+ port + " -s "+ samplerate +" -f " + freq + " -g 0"));

        startActivityForResult(intent, 1234);
    }

How do you want me to change this?

Thanks again!
Mo.

**Comment by Enthusiastic-Coder** [comment]
Hi Matioupi

Would like to know the extra arguments and their corresponding values.

Thanks.

**Comment by Enthusiastic-Coder** [comment]
Hi Matioupi

I have been looking through the Java code and I'm not sure what the argument required for me to pass via the intent. any ideas?
Thanks

**Comment by Enthusiastic-Coder** [comment]
Hi Matioupi

Do you use sdrTcpArguments tp handle the new hackRF "hackrf.setAmp(true);"

**Comment by Chef-Duck314** [comment]
All your information, Names, and darkness along with Parents is brought to light even people's job locations and everything I've gotten so far my partner isnt a good lookout !!! P. S
It's not good to whistle in the dark brings out bad spirts!! God bless you!!! B
